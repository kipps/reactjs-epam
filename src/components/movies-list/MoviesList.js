import React from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";
import store from '../../redux/store/configureStore'
import {fetchPostsRequest, searchByTitle, headerSearchSet} from "../../redux/actions/MoviesAction";
import Movie from '../movie/Movie';
import Container from 'react-bootstrap/Container';
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";


class MoviesList extends React.Component {
    componentDidMount() {
        let path = window.location.pathname;
        path.includes('/search/') ? store.dispatch(searchByTitle(path.replace('/search/', ''))) : store.dispatch(fetchPostsRequest());
        store.dispatch(headerSearchSet(true))
    }

    componentDidUpdate(prevProps) {
        if (this.props.location.pathname !== prevProps.location.pathname) {
            console.log('route changed');
            let path = window.location.pathname;
            path.includes('/search/') ? store.dispatch(searchByTitle(path.replace('/search/', ''))) : store.dispatch(fetchPostsRequest());
            store.dispatch(headerSearchSet(true));
        }
    }

    render() {
        return (this.props.loading) ?
            <Container><h1 className='flex f-large v-center h-center'>Loading Country Names...</h1></Container> :
            (!this.props.posts.length) ?
                <Container><h1 className='f-large'> No Movies </h1></Container> :
                <Container>
                    <Row>
                        {this.props.posts.map(
                            item => (
                                <Col key={item.id} md={3}>
                                    <Movie path={'/film/'} {...item} />
                                </Col>
                            )
                        )}
                    </Row>
                </Container>
    }
}

MoviesList.propTypes = {
    posts: PropTypes.array,
    loading: PropTypes.bool,
    headerSearchShow: PropTypes.bool
}

const mapStateToProps = state => {
    return {
        posts: state.moviesState.posts,
        loading: state.moviesState.loading,
        headerSearchShow: state.moviesState.headerSearchShow
    }
}

export default connect(mapStateToProps)(MoviesList)