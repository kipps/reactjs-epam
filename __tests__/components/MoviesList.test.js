import { shallow } from 'enzyme/build';
import { configure, mount } from 'enzyme/build';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16/build';
import MoviesList from "../../src/components/movies-list/MoviesList";

configure({ adapter: new Adapter() });

describe("Radio", () => {
  let movies = {
      title: 'Begin from start',
      genre: 'Драма, Комедия',
      date: '2018',
      img: 'medium',
      key: '3432424'
    };
  let list,
      component = mount(<MoviesList movies={movies} />),
      textH1 = "No films found";


  beforeEach(() => {
    list = shallow(<MoviesList movies={movies}/>);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MoviesList movies={movies} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render correct text", () => {
    expect(
      shallow(<MoviesList movies={movies} />).find("h1")
        .first()
        .text()
    ).toEqual(textH1);
  });

});
