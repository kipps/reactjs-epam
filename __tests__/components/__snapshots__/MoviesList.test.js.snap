// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1
        className="f-large"
      >
        No films found
      </h1>
    </div>
  </div>
</div>
`;
